### ISSUE TITLE
Add versioning to setup.py requirements

### ISSUE THREAD

**Issue by brettpthomas** [segmentio/analytics-python#58]
I had a previous version of python-dateutil (1.5) installed from another library, which broke `pip install analytics-python`.

**Comment by dcc635** [comment]
We also ran into this issue. We freeze our python-dateutil version which we import in our application to make sure nothing breaks if a new version is released. Because analytics-python always uses the latest version of their dependencies, that conflicts with our frozen version. You can read more about the conundrum we are in [here](http://blog.miguelgrinberg.com/post/the-package-dependency-blues). Please add versioning to this ASAP as we don't want our code to break now or in the future!

**Comment by calvinfo** [comment]
Makes sense to me–what version are you all running? Locally I have the
following.

```
python-dateutil==1.5
requests==2.7.0
six==1.4.1
```

Let me know and I can update the lib!

**Comment by dcc635** [comment]
Hi and thanks for the response (your support team was a bit less helpful FYI). Here are our current versions (we've already changed our python-dateutil version to match yours).
```
python-dateutil==2.5.0
requests==2.5.1
six==1.10.0
```
We're on analytics-python==0.4.4. Please let me know when you've published a release with versioning and we'll update to the latest version of analytics-python.

Thank you!

**Comment by calvinfo** [comment]
Okay, they're pinned now in `1.2.0`. Happy to accept PRs to loosen the requirements!

**Comment by calvinfo** [comment]
Erp, one sec. Missing python3 deps, I'll fix it now.

**Comment by calvinfo** [comment]
Okay, actually fixed and passing in `1.2.1`. Happy to relax requirements as well as long as tests pass. Thanks all!

**Comment by dcc635** [comment]
Thanks for the help so far. If I'm not mistaken though, I believe this will not solve the problem for us in the future. The reason we ran into our issue is that there was no upper bound on the versions. So when we installed analytics-python in our build, it installed the latest and greatest of these packages for us, which our build wasn't ready for (we use these same packages elsewhere in our code, especially `requests`). There needs to be an upper bound on all of these requirements, else the same situation will occur in the future. I can submit a PR if need be.
